chapters/10_chapter_9.tex

```diff
@@ -0,0 +1,329 @@
+In a clock glitching attack the attacker would inject a rising edge on the clock bus at an arbitrary time. This way only \emph{some} of the computations will have finished by that time while others are still being processed, and thus the device will be in a faulty (unstable) state.
+
+A notable example is the attack on Mifare Classic RFID chips we talked about in the beginning of the course \cite{bib:nohl2008} -- the RNG in the chip is only dependent on the time between powering up the RFID tag and challenging it. An RFID reader operated by the attacker can control both parameters, thus making the generated challenges deterministic.
+
+\subsubsection{Temperature attacks}
+This attack method relies on the physical property of electrons (current). Electrons  "jump", and the higher the temperature -- they will jump more frequently and to longer distances.
+If a device gets \emph{too hot} -- enough electrons can "jump" e.g. over the insulation layer in a transistor to flip it from logical 1 to 0. This results in a fault.
+
+Because of the ubiquity of devices failures due to temperature, nowadays temperature sensors are integrated into most devices, so when it overheats -- the device will shut down.
+
+An attacker can bypass the temperature sensor by disconnecting it. Another method would be to quickly alternate the temperature of the device from extremely high to extremely low, so that \emph{on average} the temperature is reasonable, but it will still experience faults during the extreme phases of the cycles.
+
+In an interesting research paper \cite{bib:appel} a type-confusion attack on the Java virtual-machine was demonstrated: at first, the entire memory was filled with small arrays (say of size one). The Java VM is type-safe, so it is normally impossible to access one of the memory regions using a pointer to a different region. To inject a type-confusion fault the researchers used a 50W lightbulb to heat the memory chip of the device enough to flip some of the bits (for illustration see Fig. \ref{fig:memory_lightbulb}). As a result, a small portion of the data-structures describing the arrays in memory now held wrong values (e.g. changed their value from $size=1$ to $size=20$). At this point, some affected data-structure \emph{contains} a header of a different data-structure, to which the attackers now have read and write access. Changing the header of the second data structure to an arbitrary value gave the attackers access to the entirety of the device's memory.
+\begin{figure}[!ht]
+    \centering
+	\includegraphics[width=0.7\linewidth]{images/ch9/bulb.png}
+	\caption{A lightbulb flipping memory bits filled with safe Java structures.}
+	\label{fig:memory_lightbulb}
+\end{figure}
+
+\subsubsection{Optical, Electromagnetic}
```

Comment:
Please label each section, subsection, etc. Use the convention {sec:...} or {subsec:...}, etc.

Reply:
Done